src/mediaEditor/plugins/mediaAlignment/helper.js

```diff
@@ -22,29 +22,6 @@ export const FLOAT_LEFT_CLASS = 'wrap-left';
 export const FLOAT_RIGHT_CLASS = 'wrap-right';
 export const CENTER_CLASS = 'tao-centered';
 
-const searchRecurse = (parentElement, serial) => {
-    if (!parentElement) {
-        return null;
-    }
-    if (parentElement.serial === serial) {
-        return parentElement;
-    }
-    let found = null;
-    _.some(parentElement['elements'], childElement => {
-        if (childElement.serial === serial) {
-            found = parentElement;
-        } else if (childElement['elements']) {
-            found = searchRecurse(childElement, serial);
-        } else if (childElement['prompt']) {
-            found = searchRecurse(childElement.prompt.bdy, serial);
-        }
-        if (found) {
-            return true;
-        }
-    });
-    return found;
-};
```

Comment:
Tagging @marpesia to see if she remembers something about the problem this code was added to solve 2 years ago.

Reply:
The code is working well on Authoring.
It rings a bell that it was done to be backwards compatible with previous items. This code was implemented at the end of 2022. Maybe it won't be an issue anymore.
I couldn't find any test exported from that time to validate my thoughts.

Reply:
I found the same code on itemqti. I think it is to replace any old item with the Figure tag. https://github.com/oat-sa/extension-tao-itemqti/pull/2279/commits/ba70db0af1f5e8c9cac665bf8a9fff45acdba601

Regarding the refresh() part, I remember we had some issues updating the CKEditor and keeping the changes.

Reply:
Thanks! I checked it again and found at least one difference: in A-block, when `figure` is added in the middle of text inside `<p>`, the content of `<p>` is split to mutliple lines ('`<figure>` is not allowed inside `<p>`? And Prompt or TextReader use `<div>` not `<p>`). 
In my version, user wouldn't see that until he focuses text in A-block again, or reopens the item. 

![Screenshot_21](https://github.com/user-attachments/assets/d4179dd6-61eb-44ef-9a19-9b704f7b28aa)

I'll restore old version, and add TextReader PCI check. Not sure how to check for A-block. As for `contentModel`, both A-block and TextReader have `contentModel='itemBody'`. And `.qti-flow-container` in GapMatch/Hottext doesn't have this issue.

Reply:
Regarding the A-block, I tested on oat-dev to see differences, and the behaviour is the same. 
`<Figure>` is a block time, so it can't be inside a `<p>`.
I think it is ok.